File: print_d_triangle.js
const printDTriangle = (row) => {
    for(let i = 0; i < row; i++){
        process.stdout.write('*');
        if(i>0){
            process.stdout.write(' '.repeat(2*i-1) + '*');
        }
        console.log();
    }

    for (let j = row; j >= 0; j--) {
        process.stdout.write('*');
        if (j >= 1) {
            process.stdout.write(' '.repeat(j*2-1) + '*');
        }
        console.log();
    }
}
printDTriangle(row);

File: print_d_triangle.py
'''
*
* *
*   *
*     *
*       *
*         *
*           *
*         *
*       *
*     *
*   *
* *
*
'''
def print_d_triangle(row):
    for i in range(0, row):
        print('*', end='')
        if(i>0):
            print(' '*(2*i-1), '*', end='') 
        print()

    for j in range(row, -1, -1):
        print('*', end='')
        if(j>=1):
            print(' '*(j*2-1), '*', end='')
        print() 
        
row = int(input())
print_d_triangle(row)

